File: src/main/java/com/jss/osiris/modules/osiris/reporting/service/ReportingDashboardService.java
package com.jss.osiris.modules.osiris.reporting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.reporting.model.ReportingDashboard;

public interface ReportingDashboardService {
    public List<ReportingDashboard> getReportingDashboards();

    public ReportingDashboard getReportingDashboard(Integer id);

    public ReportingDashboard addOrUpdateReportingDashboard(ReportingDashboard reportingDashboard);

    public ReportingDashboard getReportingDashboardById(Integer id);

    public List<ReportingDashboard> getReportingDashboardsForCurrentUser();
}

File: client/src/app/modules/reporting/services/reporting.dashboard.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ReportingDashboard } from '../../reporting/model/ReportingDashboard';

@Injectable({
  providedIn: 'root'
})
export class ReportingDashboardService extends AppRestService<ReportingDashboard>{

  constructor(http: HttpClient) {
    super(http, "reporting");
  }

  getReportingDashboards() {
    return this.getList(new HttpParams(), "reporting-dashboards");
  }
  
   addOrUpdateReportingDashboard(reportingDashboard: ReportingDashboard) {
    return this.addOrUpdate(new HttpParams(), "reporting-dashboard", reportingDashboard, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
